python_modules/dagster/dagster/core/log_manager.py

```diff
@@ -148,12 +148,49 @@ def construct_log_string(
     )
 
 
+class PythonLogCaptureHandler(logging.Handler):
+    def __init__(self, level: int, log_manager: "DagsterLogManager"):
+        self._log_manager = log_manager
+        super().__init__(level=level)
+
+    def _extract_extra(self, record: logging.LogRecord) -> Dict[str, Any]:
+        # hacky way to figure out which attributes were added into the record object (i.e. were
```

Comment:
It took me a couple reads to understand why this function was needed, adding more context to this comment describing where attributes are added to the extra dict might be helpful, since the behavior of this function is coupled with the fields added in `.log`.